WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartProductUpdater.kt

```diff
@@ -0,0 +1,166 @@
+package com.woocommerce.android.ui.woopos.home.cart
+
+import com.woocommerce.android.R
+import com.woocommerce.android.ui.woopos.home.ChildToParentEvent
+import com.woocommerce.android.ui.woopos.home.ParentToChildrenEvent
+import com.woocommerce.android.ui.woopos.home.WooPosChildrenToParentEventSender
+import com.woocommerce.android.ui.woopos.util.format.WooPosFormatPrice
+import com.woocommerce.android.viewmodel.ResourceProvider
+import javax.inject.Inject
+
+class WooPosCartProductUpdater @Inject constructor(
+    private val childrenToParentEventSender: WooPosChildrenToParentEventSender,
+    private val resourceProvider: ResourceProvider,
+    private val formatPrice: WooPosFormatPrice
+) {
+    suspend operator fun invoke(
+        currentState: WooPosCartState,
+        updatedProducts: List<ParentToChildrenEvent.OrderCreated.ProductInfo>,
+    ): WooPosCartState {
+        val currentBodyState = currentState.body as? WooPosCartState.Body.WithItems ?: return currentState
+        val mutableCurrentBodyList = currentBodyState.itemsInCart.toMutableList()
+        var changesDone = false
+
+        val availableProductsMap = createAvailableProductsMap(updatedProducts)
+
+        currentBodyState.itemsInCart.forEachIndexed { index, item ->
+            when (item) {
+                is WooPosCartItemViewState.Product -> {
```

Comment:
```suggestion
                is WooPosCartItemViewState.Product.Variation -> {}
                is WooPosCartItemViewState.Product.Simple -> {
```
Can we make the code more explicit or do you think this code will work for variations too when the backend is updated?

Reply:
The purpose of this code is to work with "product," and I believe that the fact the backend behaves differently for certain types of products should not influence our part. Or wdyt?